### ISSUE TITLE
ReazonSpeech FastConformer encoder runs CPU-only (RTF ~3.5 on flagship Android) — wire NNAPI/QNN heterogeneous encoder

### ISSUE THREAD

**Issue by m96-chan** [m96-chan/any-stt#16]
## Summary

The ReazonSpeech backend runs its FastConformer encoder **entirely on CPU**, even on flagship mobile SoCs, giving **RTF ≈ 3.5** (3.5× slower than realtime). The `nnapi` / `qnn` Cargo features are empty stubs, and the (functional) `nnapi-backend` / `qnn-backend` crates are Whisper-only and are never called from any `SttEngine::transcribe` path. To reach RTF < 1 for realtime voice-to-voice on-device, the FastConformer encoder needs a genuine NPU (NNAPI / QNN-HTP) implementation and a dispatch from `reazonspeech-backend`.

## Measured data (real devices, `japanese_test.wav`, 7.37 s, 16 kHz)

Standalone probe calling `reazonspeech_backend::initialize()` + `engine.transcribe()`; transcription is correct on all runs (`吾輩は猫である…`).

| Device | SoC | Quant | infer time | **RTF** |
|---|---|---|---|---|
| Pixel 10a | Tensor G4 | F16 | 27.3 s | **3.70** |
| Nubia (SM8650) | Snapdragon 8 Gen 3 | F16 | 25.9 s | **3.51** |
| Nubia (SM8650) | Snapdragon 8 Gen 3 | **Q8_0** | 25.6 s | **3.47** |
| Redmi (MT6899) | Dimensity 8400 | F16 | 21.9 s | **2.97** |

Notably **Q8_0 is no faster than F16** — `fastconformer-core` dequantizes to f32 and runs the same NEON path, so quantization saves memory but not compute. `active_backend()` reports `nnapi`, but that is a cosmetic label; the work is 100% CPU.

## Root cause (rev `84d8655`)

- `reazonspeech-backend/src/lib.rs::transcribe()` calls `self.encoder.forward(&mel, n_frames)` unconditionally; `self.encoder` is a pure-CPU `fastconformer_core::encoder::FastConformerEncoder` (deps: `realfft`, `rayon`, `half` — no NPU/GPU).
- `self.backend` (from `selector::select`) only feeds `SttResult.backend_used` / `active_backend()`; it never branches execution.
- `reazonspeech-backend/Cargo.toml` features `qnn = []`, `nnapi = []` are empty and the crate does **not** depend on `nnapi-backend` / `qnn-backend`.
- `nnapi-backend` (real `dlopen` of `libneuralnetworks.so`, model compile/execute) and `qnn-backend` (real `libQnnHtp.so` graph builder) implement the **Whisper** transformer encoder only, and are driven **only** by `crates/bench` — no `SttEngine` reaches them.

## Proposed implementation

1. **Cargo wiring**: `reazonspeech-backend/Cargo.toml`
   ```toml
   nnapi = ["dep:nnapi-backend"]
   qnn   = ["dep:qnn-backend"]
   ```
2. **New FastConformer NPU encoders** in `nnapi-backend` / `qnn-backend`, mirroring the existing `WhisperEncoder*` but for the FastConformer block. The CPU interface to match is:
   `FastConformerEncoder::forward(&self, mel: &[f32], n_frames: usize) -> EncoderOutput { data: Vec<f32>, n_frames, d_model }`.
   FastConformer ops the NPU graph must cover:
   - Conv2d subsampling ×8 (two stride-2 convs) on the log-mel `[T, n_mels]`.
   - Macaron feed-forward (two half-step FFNs) — **reuse** the existing Whisper FFN/GELU graph nodes.
   - Self-attention with **relative** positional encoding (Transformer-XL `pos_bias_u` / `pos_bias_v`) — the existing Whisper MHA is absolute-position; the rel-pos bias term is the missing piece.
   - Conv module: pointwise → GLU → depthwise-conv → batchnorm → SiLU/swish → pointwise (no counterpart in the Whisper graph).
   - LayerNorm / matmul / softmax — **reuse** existing nnapi/qnn ops.
3. **Dispatch** in `reazonspeech-backend::transcribe()`: when `self.backend` is `Nnapi`/`Qnn` and the feature is compiled, run `FastConformerEncoderNnapi/Qnn::execute(&mel)` instead of `self.encoder.forward(...)`; fall back to CPU on any error (fail-open).
4. Keep the RNN-T / TDT decoder on CPU (small; `decoder.rs` already notes NPU offload does not pay off there).
5. Close the QNN gaps that block correctness even for Whisper today: dropped Q/out/FC1/FC2 biases (`qnn-backend/src/encoder.rs:360/422/464/488`) and `SplitStrategy::HeadSplit` (`:142`).

## Verification harness

A ready-to-use RTF probe (loads the `.gguf`, transcribes a wav, prints RTF + backend, runs on `adb shell` since ReazonSpeech is CPU-only today) lives in the consumer repo at `e2e/stt_workload/`. Re-running it after the NPU encoder lands should show RTF < 1 with `active_backend=nnapi`/`qnn` reflecting real offload.

This is a sizable feature (a new FastConformer NPU graph), not a flag flip. Happy to help scope/implement it in stages (start with NNAPI on Snapdragon/Tensor, then QNN-HTP).
